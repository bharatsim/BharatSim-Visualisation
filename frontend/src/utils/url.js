const DATASOURCE_URL = '/api/dataSources';
const DASHBOARD_URL = '/api/dashboard';
const PROJECT_URL = '/api/projects';

export const url = {
  getDataUrl: (dataSource) => `${DATASOURCE_URL}/${dataSource}`,
  getHeaderUrl: (dataSource) => `${DATASOURCE_URL}/${dataSource}/headers`,
  getProjectUrl: (projectId) => `${PROJECT_URL}/${projectId}`,
  getDashboardUrl: (dashboardId) => `${DASHBOARD_URL}/${dashboardId}`,
  INSERT_DASHBOARD: `${DASHBOARD_URL}/create-new`,
  DATA_SOURCES: `${DATASOURCE_URL}`,
  DASHBOARD_URL,
  PROJECT_URL,
};
