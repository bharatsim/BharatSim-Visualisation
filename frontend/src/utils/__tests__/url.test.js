import { url } from '../url';

describe('URL', () => {
  it('should provide a url', () => {
    expect(url).toMatchSnapshot();
  });

  it('should provide a url for headers', () => {
    expect(url.getHeaderUrl('dataSource')).toEqual('/api/dataSources/dataSource/headers');
  });

  it('should provide a url for data', () => {
    expect(url.getDataUrl('dataSource')).toEqual('/api/dataSources/dataSource');
  });

  it('should provide a url for project', () => {
    expect(url.getProjectUrl('projectId')).toEqual('/api/projects/projectId');
  });
});
