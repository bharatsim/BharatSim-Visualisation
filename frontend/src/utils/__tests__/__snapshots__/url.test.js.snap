// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`URL should provide a url 1`] = `
Object {
  "DASHBOARD_URL": "/api/dashboard",
  "DATA_SOURCES": "/api/dataSources",
  "INSERT_DASHBOARD": "/api/dashboard/create-new",
  "PROJECT_URL": "/api/projects",
  "getDashboardUrl": [Function],
  "getDatasourceUrl": [Function],
  "getHeaderUrl": [Function],
  "getProjectUrl": [Function],
}
`;
